webrecorder/browsertrix
# Configure whether frontend sign up route is enabled

Uses build-time environment variable to determine whether `/sign-up` route should exist.

TODO need to replace with API route to dynamically pull this value at build time. For now, the `.env` file needs to be manually updated.

### Manual testing
1. Add `REGISTRATION_ENABLED=true` to `.env.local`
2. Run `yarn start-dev` and go to http://localhost:9870/sign-up. You should see the sign up form
3. Stop dev server and change `REGISTRATION_ENABLED` to false or remove.
4. Restart dev server and go to http://localhost:9870/sign-up. You should see the default 404 page

### Opinions
I chose to move this to a build-time variable so that:
- the sign up page isn't blocked by an API request on every page load
- ease of overriding for development (update .env.local file)

I figure enabling/disabling the page won't happen very often, so it wouldn't be too much of a process to re-deploy the frontend when registrations are enabled.
